src/main/java/com/datatorrent/netlet/AbstractClient.java

```diff
@@ -337,7 +391,7 @@ public boolean send(byte[] array, int offset, int len)
   @Override
   public void handleException(Exception cce, EventLoop el)
   {
-    logger.debug("Collecting exception in {}", throwables.size(), cce);
+    logger.error("Exception in event loop {}", el, cce);
```

Comment:
It is based on `verifyExceptionsInClientCallback()`. Please see TODO comments. In the use case where client requests connection to be established, sends a message and waits for a response, it may never get a response and there will be no exception thrown if send is called before actual connection is established and something goes wrong during connect.